core/class/CommandLineTool.go

```diff
@@ -185,6 +190,41 @@ func (tool *CommandLineTool) placeInputFilesToCommandExecDir() error {
 	return nil
 }
 
+// defineStdoutDestination
+func (tool *CommandLineTool) defineStdinSource() error {
+	if tool.Root.Stdin != "" {
+		stdin, err := tool.Command.StdinPipe()
+		if err != nil {
+			return err
+		}
+		defer stdin.Close()
+		stdinfilename := tool.Root.Stdin
+		if strings.HasPrefix(stdinfilename, "$(") && strings.HasSuffix(stdinfilename, ")") {
```

Comment:
if it become `${...}` in the future, it's difficult to figure out where we should modify. Thus, we better create a method like `NeedVariableEvaluation(s string) bool` to separate this long condition.